### PR TITLE
[13.0][MIG] account_journal_lock_date: Migration to 13.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
cc @Tecnativa TT26071

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Code and functional test :+1:

File: account_journal_lock_date/__manifest__.py [hunk 1]
```diff
@@ -5,11 +5,11 @@
     "name": "Account Journal Lock Date",
     "summary": """
         Lock each journal independently""",
-    "version": "12.0.2.0.0",
+    "version": "13.0.1.0.0",
     "license": "AGPL-3",
-    "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
+    "author": "ACSONE SA/NV, Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline account_journal_lock_date/__manifest__.py]: Shouldn't we be here after the major feature we added?

File: account_journal_lock_date/models/account_move.py [hunk 2]
```diff
@@ -15,18 +16,17 @@ class AccountMove(models.Model):
     @api.model
     def create(self, values):
         move = super().create(values)
-        move._check_lock_date()
+        move._check_fiscalyear_lock_date()
```
Comment (pedrobaeza) [inline account_journal_lock_date/models/account_move.py]: Is it needed to call explicitly this method? Isn't done by standard?

File: account_journal_lock_date/__manifest__.py (hunk 1 above)
Comment (ernestotejeda) [reply account_journal_lock_date/__manifest__.py]: Yes, I think we should. Done

File: account_journal_lock_date/models/account_move.py (hunk 2 above)
Comment (ernestotejeda) [reply account_journal_lock_date/models/account_move.py]: In the create of account.move, that method is not called, but it is called from the create method of account.move.line, so I think we could remove the override of the create method.

And in the override of the write method I left only one _check_fiscalyear_lock_date call

**Review by pedrobaeza — APPROVED** [review]

**Review by luc-demeyer — COMMENTED** [review]
I tend to install following module as an alternative to this OCA module:
https://github.com/Noviat/noviat-apps/tree/13.0/account_journal_lock_date_nov

Main reason is that if my customers want to protect a journal, than this protection should also be for the advisors.

I also do not encounter a fiscalyear_lock_date requirement on the level of journal, hence as a consequence my module has only a few lines of code and does the job (although I intend to add the wizard unless if we manage to come to common approach  (e.g. config setting to set the default 'advisor' protection level).

**Comment by pedrobaeza** [comment]
So now this module complies with all your requirements or there is something missing that you want to add here?

**Comment by luc-demeyer** [comment]
@pedrobaeza 
this OCA module allows the advisor to post entries in a locked journal which is the main reason why I maintain an own version. That's why I suggested in my comments to add a config setting param for this.

**Comment by pedrobaeza** [comment]
There are 2 lock dates per journal: one for advisers, and the other for non advisers, so I don't get your problem. Is a bug that the date for advisers is not working, or you are not aware of the second lock date?

**Comment by luc-demeyer** [comment]
@pedrobaeza 
Sorry. I was confused by the name 'fiscalyear_lock_date'. I now installed the module and on the User Interface the terms are clear. Why not rename the two fields to e.g lock_date and lock_date_non_advisor ?

Another difference between this module and the one I tend to use is that I only check the constraint at posting time:
cf. https://github.com/Noviat/noviat-apps/blob/83072753093ab770db77eaa74d57aad532660b3a/account_journal_lock_date_nov/models/account_move.py#L20

I think this is better but I am looking forward to reactions the other OCA financial specialists.

**Comment by pedrobaeza** [comment]
Field names follow the same nomenclature that is already in res.company in core Odoo. I know it's not the best, but it's to be consistent.

About the posting, this follows also the same behavior than in core (but not sure in v13), and I prefer it, as if you are allowed to do a change in an unposted move, but when posting it will be forbidden, you don't have a way to come back to what you get before, and also it can happen the nonsense in v8 where you unpost by mistake a move, but not being able to post it again due to the period was closed.

**Comment by luc-demeyer** [comment]
1) field names
I am not sure if we need to use same field names as in core odoo since in core odoo the 'fiscalyear_lock_date' is intended to be used for fiscal year closing whereas we use it to close a journal for all users including advisor

2) draft versus posted
core odoo uses the same logic as I do in my journal_lock_date module:
cf. https://github.com/odoo/odoo/blob/341639521ff916afa23abd2b10bbb1e9ce72ed58/addons/account/models/account_move.py#L1536

**Comment by pedrobaeza** [comment]
Odoo uses the same features for that names. It's not for fiscal year lock, but for adviser lock.

About the posting thing, I still think it's better to block it at all, for avoiding false expectations.

**Review by victoralmau — CHANGES_REQUESTED** [review]
:+1:

**Review by victoralmau — APPROVED** [review]

**Pull request merged**